pass_args_to_sdf.md

```diff
@@ -0,0 +1,201 @@
+# Pass arguments to SDF files
+
+## Overview
+
+The goal of passing arguments to SDF files is to allow a user to send custom data into a model or world file and prevent model/world file duplication. Currently, there are multiple robot models with the same base but different sensor configurations. For example, in [SubT's robots](https://github.com/osrf/subt/wiki/Robots), X1 has 8-10 different sensor configurations but all have the same base. If we take a look at all the different configurations in [Fuel](https://app.ignitionrobotics.org/), we see that they are essentially copies of each other but with different sensors or different parameters for these sensors. The aim of adding parameter support is to avoid this duplication, where a model/world file contains all configurations with default parameters and a user may update and/or override certain parameters as desired.
+
+### Features
+
+The list of intended **primary features** include:
+* Sending data from a world or launch file into a model (e.g., topic names, change parameter values in model(s))
+* Enable/disable parts of models and/or worlds
+
+**Secondary features** include:
+* Add support to [ign-fuel-tools](https://github.com/ignitionrobotics/ign-fuel-tools) and [Fuel](https://app.ignitionrobotics.org/) web application
+
+### Need to consider:
+
+* Do other libraries (e.g., [ign-physics](https://github.com/ignitionrobotics/ign-physics)) need this capability?
+
+## Implementation
+
+The initial question was whether this process should be a [SDF](http://sdformat.org/) or [ign-gazebo](https://github.com/ignitionrobotics/ign-gazebo) feature. The conclusion is that this will most likely be implemented in ign-gazebo. The reasoning behind this is other organizations use SDF with other simulators, which raises concerns that it could affect their work. Also, this desired argument passing process is intended for ign-gazebo use. Although there may be minor updates to SDFormat and/or libsdformat, the current plan is to implement the main functionality in ign-gazebo. Below describes the current idea to design this new feature.
+
+![ign-gazebo's Server current process of loading SDF](./images/ign_gazebo_curr_SDF_process.png)
+
+The figure above shows the current process, where the gazebo `Server` takes in an SDF file and libsdformat loads the SDF object. In libsdformat's `Load(filename)` function, `readFile` loads the file into `sdfParsed` which is an `SDFPtr` that contains the root `sdf::Element` as well as other information about the SDF file. Then the `Load(sdfParsed)` creates the DOM object.
```

Comment:
Could you elaborate a bit more on the advantages of this 2-step approach?

For example, another approach could be for `ign-gazebo` to load the raw XML file, make the necessary substitutions to make it into valid, pure SDF, then pass the SDF file `readFile` as a string.

Reply:
Since libsdformat already reads and parses the raw XML file which is loaded into `sdfParsed`, the advantages of using `sdfParsed` is that we can use libsdformat's API (e.g., Element class) to update elements as opposed to using tinyxml or something similar

Reply:
Oh ok got it, that sounds reasonable. What do you think of explaining that on the document for future readers?

Reply:
I agree, I added it here af03c7d. what do you think?